src/metadata.rs

```diff
@@ -254,6 +266,12 @@ impl Drop for Metadata {
     }
 }
 
+impl AsMut<grpc_metadata_array> for Metadata {
```

Comment:
How about implementing it as `pub(crate) fn as_mut_ptr(&mut self) -> *mut grpc_metadata_array`?

Reply:
I agree, this is cleaner.